frontend/app/actions/RegistrationUri.scala

```diff
@@ -33,10 +34,11 @@ object RegistrationUri {
     controllers.routes.Joiner.joinFriend().path -> "FRI"
   )
 
-  def parse(request: RequestHeader) = {
-    val redirectUrl: String = Config.idWebAppRegisterUrl(request.uri)
+  def parse(request: RequestHeader): String = {
     val campaignCode = extractCampaignCode(request.headers.get(REFERRER_HEADER), request.path)
-    redirectUrl.concat(s"&INTCMP=$campaignCode")
+    val returnUri: String = (request.uri ? ("INTCMP" -> campaignCode))
+    val redirectUrl: String = Config.idWebAppRegisterUrl(returnUri)
+    (redirectUrl ? ("INTCMP" -> campaignCode)).toString
```

Comment:
so we need the campaign code on the id url still, as well as adding it to the return url?